src/App.jsx

```diff
@@ -14,70 +15,56 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedItem, setSelectedItem] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedItem ? `${selectedItem} is selected` : 'No goods selected'}
+        {
+          selectedItem && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setSelectedItem('');
+            }}
+          />
+          )
+        }
+      </h1>
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={
+                selectedItem === good && 'has-background-success-light'
+              }
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={`${selectedItem === good ? 'RemoveButton' : 'AddButton'}`}
+                  type="button"
+                  className={`${selectedItem === good ? 'button is-info' : 'button'}`}
+                  onClick={() => {
+                    setSelectedItem(selectedItem === good ? '' : good);
+                  }}
+                >
+                  {`${selectedItem === good ? '-' : '+'}`}
+                </button>
```

Comment:
if you change almost every prop depending on that condition, it is better to write
 ```
{selectedItem === good 
 ? (<button ...>...</button>)
  : (<button ...>...</button>)
```